src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -693,7 +704,7 @@ protected virtual JwtSecurityToken ValidateSignature(string token, TokenValidati
                 {
                     try
                     {
-                        if (ValidateSignature(encodedBytes, signatureBytes, securityKey, jwt.Header.Alg))
+                        if (ValidateSignature(encodedBytes, signatureBytes, securityKey, algorithm))
```

Comment:
I'm fine moving the JWA -> XMLDSIG mapping to the private `ValidateSignature` method if you think it's clearer, but what's sure is that we'll need to call `GetSignatureProviderForValidating` with the XMLDSIG identifier, not with the JWA short name.